.github/workflows/actions.yml

```diff
@@ -45,42 +47,44 @@ jobs:
         with:
           arguments: --refresh-dependencies --stacktrace --scan clean build
       - name: Sonarqube
-        if: matrix.java == '8' && matrix.os == 'ubuntu-latest'
+        if: matrix.java == '8' && matrix.os == 'ubuntu'
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           SONAR_TOKEN: "61ab2579215aa8a0024a2f9368fc1298fdecfd18"
         run: ./gradlew jacocoTestReport sonarqube --stacktrace -i
 
-  # This will be handled on a later stage
-  # deploy:
-  #   runs-on: ubuntu-latest
-  #   name: Deploy
-  #   if: github.ref == 'refs/heads/master'
-  #   steps:
-  #     - name: Check out repo
-  #       uses: actions/checkout@v2
-  #     - name: Setup java
-  #       uses: actions/setup-java@v1
-  #       with:
-  #         java-version: 8
-  #     - name: Cache Gradle
-  #       uses: actions/cache@v1
-  #       with:
-  #         path: ~/.gradle/caches
-  #         key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
-  #         restore-keys: ${{ runner.os }}-gradle-
-  #     - name: Perform Release
-  #       uses: eskatos/gradle-command-action@v1
-  #       if:
-  #       with:
-  #         arguments: ciPerformRelease
-  #     - name: Upload Release to GitHub
-  #       uses: softprops/action-gh-release@v1
-  #       if: startsWith(github.ref, 'refs/tags/')
-  #       with:
-  #         files: ./build/libs/junit-pioneer-*.jar
-  #       env:
-  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  deploy:
```

Comment:
If we want to, we can also use a separate YAML file for this?

Reply:
i was also thinking about this, and maybe trigger this from the build script. but i think we should always first run the build, and see if there are no errors, and than deploy.

This might be a good improvement for the future to find a sophisticated way of running both actions in sequence but conditionally on a success of the first one. I did not investigate that far into actions.